js/filter-tags.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+var $ = require('jquery');
+var _ = require('underscore');
+
+var BODY_TEMPLATE = _.template(
+  '<div class="data-container__tags">' +
+    '<h3 class="tags__title">Viewing: ' +
+      '<span class="js-tag-title tags__title__text">{{ title }}</span>' +
+    '</h3>' +
+    '<ul class="tags"></ul>' +
+  '</div>',
+  {interpolate: /\{\{(.+?)\}\}/g}
+);
+
+var TAG_TEMPLATE = _.template(
+  '<li data-id="{{ key }}" class="tag">' +
+    '{{ value }}' +
+    '<button class="button js-close tag__remove">' +
+      '<span class="u-visually-hidden">Remove</span>' +
+    '</button>' +
+  '</li>',
+  {interpolate: /\{\{(.+?)\}\}/g}
+);
+
+function TagList(opts) {
+  this.opts = opts;
+
+  this.$body = $(BODY_TEMPLATE({title: this.opts.title}));
+  this.$list = this.$body.find('ul');
+  this.$title = this.$body.find('.js-tag-title');
+
+  $(document.body)
+    .on('filter:added', this.addTag.bind(this))
+    .on('filter:removed', this.removeTagEvt.bind(this))
+    .on('filter:renamed', this.renameTag.bind(this));
+
+  this.$list.on('click', '.js-close', this.removeTagDom.bind(this));
+}
+
+TagList.prototype.addTag = function(e, opts) {
+  this.removeTag(opts.key, false);
+  this.$title.html('');
+
+  var $tag = TAG_TEMPLATE(opts);
```

Comment:
Small nitpick:  looks like this could be moved to the beginning of this method (accounting for JS variable hoisting and code readability/consistency).